app/controllers/api/v1/rooms_controller.rb

```diff
@@ -59,6 +59,11 @@ def public_show
 
       # POST /api/v1/rooms.json
       def create
+        return render_error status: :bad_request, errors: 'RoomLimitError' unless PermissionsChecker.new(
```

Comment:
IK we've asked for this,
But, IMO the permission checker is used more like a private method where it can change its signature in the future supporting new permissions,
It's expected to be used through the `ensure_authroized`,
This introduces both inconsistency, requires more boiler plate code and expands the size of the action,
Future contributors can misuse the permission checker easily,